iothub/device/src/Transport/AmqpIot/AmqpIotTransport.cs

```diff
@@ -75,103 +71,37 @@ internal async Task<TransportBase> InitializeAsync(CancellationToken cancellatio
                 Logging.Enter(this, nameof(InitializeAsync));
 
             TransportBase transport;
+            AmqpTransportInitiator amqpTransportInitiator;
 
             switch (_amqpTransportSettings.Protocol)
             {
                 case IotHubClientTransportProtocol.Tcp:
-                    var amqpTransportInitiator = new AmqpTransportInitiator(_amqpSettings, _tlsTransportSettings);
-                    transport = await amqpTransportInitiator.ConnectAsync(cancellationToken).ConfigureAwait(false);
+                    amqpTransportInitiator = new AmqpTransportInitiator(_amqpSettings, _tlsTransportSettings);
                     break;
 
                 case IotHubClientTransportProtocol.WebSocket:
-                    transport = _clientWebSocketTransport = (ClientWebSocketTransport)await CreateClientWebSocketTransportAsync(cancellationToken)
-                        .ConfigureAwait(false);
+                    var websocketUri = new Uri($"{Scheme}{_hostName}:{SecurePort}{UriSuffix}");
+                    var websocketTransportSettings = new WebSocketTransportSettings
+                    {
+                        Uri = websocketUri,
+                        Proxy = _amqpTransportSettings.Proxy,
+                        SubProtocol = Amqpwsb10,
```

Comment:
you'll also want to set the certificate info in here for authentication, right? Or, is that available and being used from the amqp settings?
https://github.com/Azure/azure-iot-sdk-csharp/pull/2805/files#diff-fac10b4503d13d7e8f12936f778d1e1c1701bc952ea18bad928f5455ddce923bL161

Reply:
certificate field is not available in the AMQP library for WebSocketTransportSettings.
Weirdly, it is available for TlsTransportSettings and not WebSocketTransportSettings

Reply:
And authentication still succeeds?

Reply:
It does not, I was looking into failing tests, didn't know it was due to certificate